File: Java/XSS.java
import javax.servlet.http.*;
import javax.servlet.*;
import java.io.*;

public class XSS extends HttpServlet {
    public void doGet(HttpServletRequest request, HttpServletResponse response) 
            throws ServletException, IOException {
        
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        
        // XSS vulnerability - user input directly written to response without encoding
        String userInput = request.getParameter("name");
        
        out.println("<html><body>");
        out.println("<h1>Welcome " + userInput + "</h1>");
        out.println("</body></html>");
    }
}

File: JavaScript/xss.js
const express = require('express');
const app = express();

app.get('/welcome', (req, res) => {
    const name = req.query.name;
    
    // XSS vulnerability - user input directly rendered without sanitization
    res.send(`<html><body><h1>Welcome ${name}</h1></body></html>`);
});

app.get('/search', (req, res) => {
    const searchTerm = req.query.q;
    
    // XSS vulnerability - reflected XSS
    res.send(`
        <html>
        <body>
            <h1>Search Results</h1>
            <p>You searched for: ${searchTerm}</p>
        </body>
        </html>
    `);
});

app.listen(3000, () => {
    console.log('Server running on port 3000');
});
